frontend/services/src/filterService.ts

```
@@ -128,3 +136,129 @@ export function filterEventsBySport(eventDataList: EventData[], sportType: strin
+ * Returns a filtered list of up to 15 events from the firebase storage.
+ *
+ * @param filterFieldsMap: Map where the key is the event field name as seen in firebase.
+ * See full documentation here: https://owenyang.atlassian.net/wiki/spaces/SD/pages/20414465/Firebase+Event+Filtering
+ * @returns EventData[]
+ */
+export async function filterEvents(filterFieldsMap: { [key: string]: any }) {
+  const whereClauseList: QueryFieldFilterConstraint[] = [];
+  Object.keys(filterFieldsMap).forEach(async (key: string) => {
+    switch (key) {
+      case "startDate":
+        const startDate: Timestamp = filterFieldsMap["startDate"].startDate;
```

Biome (2.5.5)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)